spec.md

```diff
@@ -3938,6 +3952,27 @@ The shift right operation result is the arbitrary precision signed integer arith
 e2 sign bits from e1 are shifted into the most significant bits, and the e2 least significant bits of e1 are truncated.
 e2 must be non-negative.
 
+## List Operations
+
+List operations create `List`{.firrtl} property type expressions from other property expressions.
+
+### List Construction Operation
+
+  Name        Arguments   Arg Types   Result Type
+  ----------- ----------- ----------- -------------
+  List\<t\>   (e\*)       (t\*)       List\<t\>
```

Comment:
This section breaks a lot of invariants previously held by primops. Is it even appropriate to categorize it as one?

For example, you gave `List<t>` special syntax, thanks to the generic. Every other primop is signified with a simple keyword. Is `List` the primop? Or are the primops `List<Integer>`, `List<List<Integer>>`, `List<String>`, etc.

No other previous primops in FIRRTL have been variadic. In fact, it's so *un-variadic* that is encodes the arity in the grammar. If we're going to be adding variadics (which there's an argument to be made we shouldn't, given the kind of language FIRRTL is), it warrants some commentary.

Reply:
I think I'd consider `List<T>(...)` the primop.

This is just writing down the spec as-is for a feature that has been supported in the CIRCT FIRRTL compiler for nearly a year. If we want to change it we can change it, but I think it's better to have something that has stabilized in the spec than not.